def recover_des_key(ref, faulted_outputs):
    return 0

